frontend/src/components/CompareTable.test.tsx

```diff
@@ -52,8 +56,34 @@ describe('CompareTable', () => {
     );
   });
 
+  it('logs error if xlabels and xparentlabels have different lengths', () => {
+    shallow(
+      <CompareTable
+        rows={rows}
+        xLabels={xLabels}
+        yLabels={yLabels}
+        xParentLabels={[xParentLabels[0]]}
+      />,
+    );
+    expect(consoleSpy).toHaveBeenCalledWith(
+      'Number of columns with data (3) should match the aggregated length of parent columns (2).',
```

Comment:
We try to avoid validating a console log message when possible because it is a `developer-centric` testing rather than `user-centric` testing. Can we validate the UI element instead?

Reply:
Ok, I'll use your suggestion to hide the parent labels when the aggregated length does not match the number of columsn with data, and then validate based on whether or not the parent label is present.

Reply:
Looks good! Thank you Chris!